vllm/model_executor/model_loader/loader.py

```diff
@@ -911,13 +913,42 @@ def _parse_quant_state(param_name: str,
     def _unquantized_generator(self, hf_weights_files, use_safetensors,
                                quant_state_dict) -> Generator:
         from bitsandbytes.functional import quantize_4bit
+        tp_size = get_tensor_model_parallel_world_size()
+        tp_rank = get_tensor_model_parallel_rank()
+
         for weight_name, weight_tensor in self._hf_weight_iter(
                 hf_weights_files, use_safetensors):
             if any(target_module in weight_name
                    for target_module in self.target_modules):
                 weight_name = weight_name.replace(".weight", ".qweight")
+
+                # weight partitions of different modules occur at
+                # different dimensions
+                if 'down_proj' in weight_name or 'o_proj' in weight_name:
```

Comment:
Is there anyway you could avoid matching the names? These could easily change for non-llama models

Reply:
I am actively working on supporting more models with bnb. 

The hard-coded names here, as well as the default_target_modules, which are for llama only, will be changed in the next PR in a more systematic way.

Reply:
I see, this is fine for now but it would be good to leave a TODO in the code to state intentionality

Reply:
Done.